service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ListGatekeepingHearingController.java

```diff
@@ -201,26 +252,70 @@ public CallbackResponse validateHearingDatesMidEvent(@RequestBody CallbackReques
 
     @PostMapping("validate-judge-email/mid-event")
     public CallbackResponse validateJudgeEmailMidEvent(@RequestBody CallbackRequest callbackRequest) {
-
         final CaseDetails caseDetails = callbackRequest.getCaseDetails();
         final CaseData caseData = getCaseData(caseDetails);
 
-        final JudgeAndLegalAdvisor tempJudge = caseData.getJudgeAndLegalAdvisor();
-
-        if (caseData.hasSelectedTemporaryJudge(tempJudge)) {
-            final Optional<String> error = validateEmailService.validate(tempJudge.getJudgeEmailAddress());
+        // todo - refactor me, triple nested if!!!
```

Comment:
Probably a good idea when we have a chance. :) 

Don't know if there's any shared logic between this and what's in the AllocatedJudgeController that could be split our into a helper?